### PR TITLE
add service and tasks to repair comments

### PR DESCRIPTION
**Pull request by eliegaboriau**
#### :tophat: Description
*Please describe your pull request.*

This PR adds a rake task to repair comments that are formatted like { "en" => { "en" => "foobar"}}
It turns these into { "en" => "foobar" } and does nothing to the others

#### :pushpin: Related Issues
*Link your PR to an issue*
- [Notion card](https://www.notion.so/opensourcepolitics/bd4bf860eab94f9daca7da803f09e3b2?v=c96df65755dd403d95e86f82d7749709&p=52077469164a42c68f3e170e24f54f14&pm=c)

#### Testing
*Describe the best way to test or validate your PR.*

Example:
* Change the format of a comment in the database
* run the rake task
* see it goes back to the good format

#### Tasks
- [x] Add specs

### CODE REVIEW

File: app/services/decidim/repair_comments_service.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      invalid_comments.empty?
+    end
+
+    def invalid_comments
+      @invalid_comments ||= Decidim::Comments::Comment.all.map do |comment|
+        next if translated_attribute(comment.body).is_a?(String)
+
+        comment.body.delete("machine_translations")
+        [comment, comment.body.values.first]
+      end.compact
```
Comment (armandfardeau) [inline app/services/decidim/repair_comments_service.rb:27]: The way you are doing this is suboptimal, let's peer.

**Review by armandfardeau — CHANGES_REQUESTED** [review]

File: app/services/decidim/repair_comments_service.rb (hunk 1 above)
Comment (armandfardeau) [reply app/services/decidim/repair_comments_service.rb:27]: For references: 

- https://www.rubyinrails.com/2017/11/16/use-find-each-instead-of-all-each-in-rails/

File: app/services/decidim/repair_comments_service.rb [hunk 2]
```diff
@@ -19,12 +19,16 @@ def ok?
...
 
     def invalid_comments
-      @invalid_comments ||= Decidim::Comments::Comment.all.map do |comment|
+      return @invalid_comments if @invalid_comments
+
+      invalid_comments = []
+      Decidim::Comments::Comment.find_each do |comment|
         next if translated_attribute(comment.body).is_a?(String)
 
         comment.body.delete("machine_translations")
```
Comment (armandfardeau) [inline app/services/decidim/repair_comments_service.rb:28]: Can you explain this line?

Comment (eliegaboriau) [reply app/services/decidim/repair_comments_service.rb]: It should not be here

**Pull request merged**